### PR TITLE
chore: use manifest version for redis+dex in E2E tests (#9955)

### LINKED ISSUE #9955: Add codegen to sync manifest image tags with Procfile image tags
# Summary

We need to automatically reflect changes to the redix/dex/haproxy image tags from the manifests directory to the Procfile so e2e tests use the correct version.

# Motivation

e2e tests use the processes initiated in Procfile. We should use the same version in Procfile as what is set in the manifests.

# Proposal

Probably a BASH script invoked from the Makefile. It should be part of the `codegen` target so that sync issues are caught in CI.

### PR DESCRIPTION
**Pull request by notfromstatefarm**
Signed-off-by: notfromstatefarm <86763948+notfromstatefarm@users.noreply.github.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Closes #9955

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
-dex: sh -c "ARGOCD_BINARY_NAME=argocd-dex go run github.com/argoproj/argo-cd/v2/cmd gendexcfg -o `pwd`/dist/dex.yaml && docker run --rm -p ${ARGOCD_E2E_DEX_PORT:-5556}:${ARGOCD_E2E_DEX_PORT:-5556} -v `pwd`/dist/dex.yaml:/dex.yaml ghcr.io/dexidp/dex:v2.30.2 dex serve /dex.yaml"
-redis: bash -c "if [ \"$ARGOCD_REDIS_LOCAL\" == 'true' ]; then redis-server --save '' --appendonly no --port ${ARGOCD_E2E_REDIS_PORT:-6379}; else docker run --rm --name argocd-redis -i -p ${ARGOCD_E2E_REDIS_PORT:-6379}:${ARGOCD_E2E_REDIS_PORT:-6379} redis:7.0.0-alpine --save '' --appendonly no --port ${ARGOCD_E2E_REDIS_PORT:-6379}; fi"
+dex: sh -c "ARGOCD_BINARY_NAME=argocd-dex go run github.com/argoproj/argo-cd/v2/cmd gendexcfg -o `pwd`/dist/dex.yaml && docker run --rm -p ${ARGOCD_E2E_DEX_PORT:-5556}:${ARGOCD_E2E_DEX_PORT:-5556} -v `pwd`/dist/dex.yaml:/dex.yaml $(yq e '.spec.template.spec.containers[] | select(.name == "dex").image' manifests/base/dex/argocd-dex-server-deployment.yaml) dex serve /dex.yaml"
```
Comment (alexmt) [inline Procfile:3]: Not everyone have `yq` installed. Instead, can we use `cat`, `grep`, and `cut`:

```
cat manifests/base/dex/argocd-dex-server-deployment.yaml | grep ghcr.io/dexidp/dex | cut -d':' -f3
```

 It is a little hacky but should work for most contributors.

**Review by alexmt — CHANGES_REQUESTED** [review]

File: Procfile (hunk 1 above)
Comment (notfromstatefarm) [reply Procfile]: good idea, lets cut it down to just grep and cut too! 😄 

sorry, I for some reason thought we already used yq!

**Review by alexmt — APPROVED** [review]
Thank you! LGTM

**Comment by wtam2018** [comment]
LGTM

**Pull request merged**